include/lbann/layers/learning/base_convolution.hpp

```diff
@@ -428,8 +428,13 @@ class base_convolution_layer : public learning_layer {
     // Perform transposed convolution on the GPU
     // Determine transposed convolution algorithm
     const size_t work_space_size = this->m_cudnn->get_work_space_size();
+    #ifndef LBANN_DETERMINISTIC_CUDNN
     cudnnConvolutionBwdDataAlgo_t transposed_convolution_cudnn_algorithm
       = CUDNN_CONVOLUTION_BWD_DATA_ALGO_0;
+    #else
+    cudnnConvolutionBwdDataAlgo_t transposed_convolution_cudnn_algorithm
+      = CUDNN_CONVOLUTION_BWD_DATA_ALGO_1;
+    #endif
     CHECK_CUDNN(cudnnGetConvolutionBackwardDataAlgorithm(this->m_cudnn->get_handle(),
```

Comment:
I don't think the `LBANN_DETERMINISTIC_CUDNN` flag is having an effect right now since the algorithm choice is being overwritten by `cudnnGetConvolution<Forward/Backward>Algorithm()`. This function should only be called in the non-deterministic case.

Reply:
Oops, I didn't even notice that! When I tried moving the getConvAlgo call into the non_deterministic case I get a failure in the cudnnConvolutionBackwardData call that follows. The cuDNN error is CUDNN_STATUS_BAD_PARAM. I don't know why changing the algorithm would cause this. I read through what could cause it here: https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnConvolutionBackwardData 
but unfortunately I am still unsure. Any idea what else I need to change?

Reply:
We support 3D convolutions, but I don't believe any of our current models use it. Perhaps the right answer is to use a deterministic algorithm for 2D convolutions, and just print a warning or something at startup for 3D convolutions.

Reply:
Ok, good to know. I will make those additions